website/projects/awssync.py

```diff
@@ -22,8 +26,37 @@ def button_pressed(self):
         :return: True if function executes successfully
         """
         self.logger.info("Pressed button")
+        self.logger.info(self.get_emails_with_teamids())
         return True
 
+    def get_all_mailing_lists(self):
+        """
+        Get all mailing lists from the database.
+
+        :return: List of mailing lists
+        """
+        mailing_lists = MailingList.objects.all()
+        mailing_list_names = [ml.email_address for ml in mailing_lists]
+        return mailing_list_names
+
+    def get_emails_with_teamids(self):
+        """
+        Create a tuple with email and corresponding teamID, where teamID is a concatenation of ID and semesterID.
+
+        :return: list of (email, teamid)
+        """
+        mailing_lists = MailingList.objects.all()
+        email_id = []
+        for ml in mailing_lists:
```

Comment:
You are iterating over all mailing lists, but I think it makes more sense to iterate over each project. For example (for inspiration):

```Python
from projects.models import Project
from mailing_lists.models import MailingList
from courses.models import Semester

for project in Project.objects.filter(mailinglist__isnull=False).values('name', 'semester', 'mailinglist'):
	project_name = project['name']
	project_semester = str(Semester.objects.get(pk=project['semester']))
	project_email = MailingList.objects.get(pk=project['mailinglist']).email_address
	print((project_name, project_semester, project_email))
```

Reply:
Don't think iterating over projects is the right way to go. The AWS units are made with the mailing list, so when a project has no mailing list, it is also not included in the AWS organization. That makes synchronization between the two platforms a bit easier.

Reply:
I really don't understand why you guys haven't somewhat formalized the synchronization process. I currently have no clue what the synchronization process is going to look like, making reviewing suboptimal.. I'll just handwave it away for now.